monai/transforms/io/array.py

```diff
@@ -458,15 +458,27 @@ def set_options(self, init_kwargs=None, data_kwargs=None, meta_kwargs=None, writ
             self.write_kwargs.update(write_kwargs)
         return self
 
-    def __call__(self, img: torch.Tensor | np.ndarray, meta_data: dict | None = None):
+    def __call__(
+        self, img: torch.Tensor | np.ndarray, meta_data: dict | None = None, filename: str | PathLike | None = None
+    ):
         """
         Args:
             img: target data content that save into file. The image should be channel-first, shape: `[C,H,W,[D]]`.
             meta_data: key-value pairs of metadata corresponding to the data.
+            filename: str or file-like object which to save img.
+                If specified, will ignore `self.output_name_formatter` and `self.folder_layout`.
         """
         meta_data = img.meta if isinstance(img, MetaTensor) else meta_data
-        kw = self.fname_formatter(meta_data, self)
-        filename = self.folder_layout.filename(**kw)
+        if filename is not None:
+            filename = str(filename) + (
+                f".{self.output_ext}"
+                if self.output_ext and not self.output_ext.startswith(".")
+                else f"{self.output_ext}"
+            )
```

Comment:
```suggestion
            filename = f"{filename}{self.output_ext.split('.', 1)[-1]}"
```
I think this is equivalent.

Reply:
Thanks for the suggestion, may not work for "nii.gz"?

Reply:
Actually `filename = f"{filename}{('.'+self.output_ext.lstrip('.')) if self.output_ext else ''}"` would work for that case instead. Maybe `self.output_ext` should be cleaned up in the constructor to ensure it begins with "." if not empty?

Reply:
Sure, updated it in the latest commit! Could you please review it?